### PR TITLE
Profile picture resize

### PR DESCRIPTION
**Pull request by SarahhhC**
We resize profile picture as three kinds.

small => 100
meduim(default) => 300
large => 600

@front For chatting profile use small and for mentorlist use large etc...

### CODE REVIEW

File: src/server/controllers/users.controller.js [hunk 1]
```diff
@@ -19,7 +19,14 @@ const ObjectId = mongoose.Types.ObjectId;
...
 const platform = { local: '0', facebook: '1', linkedin: '2' };
-const defaultProfileUrl = 'https://s3.ap-northeast-2.amazonaws.com/yodabucket/profile/default/pattern.png';
+
+// Image constant vars.
+const bucketName = 'yodabucket';
+const IMAGE_SIZE_SMALL = '100';
+const IMAGE_SIZE_MEDIUM = '300';
+const IMAGE_SIZE_LARGE = '600';
+const S3_endpoint_href = `https://s3.ap-northeast-2.amazonaws.com/`;
+const defaultProfileUrl = `${S3_endpoint_href}${bucketName}/profile/default/pattern`;
```
Comment (MechanicKim) [inline src/server/controllers/users.controller.js]: Identifier 'S3_endpoint_href' is not in camel case

File: src/server/controllers/users.controller.js [hunk 2]
```diff
@@ -410,23 +421,40 @@ function crawlByAccessTokenFacebook(accessToken) {
...
-            method: 'GET',
-            url: FB_GRAPH_BASE_URL + (result.id + '/') + FB_GRAPH_GET_PICTURE_URI,
-            qs: { type: 'large', redirect: '0' },
-            resolveWithFullResponse: true,
-          });
+          return crawlFacebookProfileBySize(result.id, IMAGE_SIZE_SMALL);
         }
       })
-      .then((facebookPictureResult) => {
+      .then((facebookSmallPictureResult) => {
```
Comment (MechanicKim) [inline src/server/controllers/users.controller.js]: Expected to return a value at the end of this function(line 437 ~ 438)

File: src/server/controllers/users.controller.js [hunk 3]
```diff
@@ -410,23 +421,40 @@ function crawlByAccessTokenFacebook(accessToken) {
...
+          if (JSON.parse(facebookSmallPictureResult.body).data.is_silhouette) {
+            result.profile_picture_small = `${defaultProfileUrl}_small`;
+          } else {
+            result.profile_picture_small = JSON.parse(facebookSmallPictureResult.body).data.url;
+          }
+
+          return crawlFacebookProfileBySize(result.id, IMAGE_SIZE_MEDIUM);
+        }
+      })
+      .then((facebookPictureResult) => {
```
Comment (MechanicKim) [inline src/server/controllers/users.controller.js]: Expected to return a value at the end of this function(line 448 ~ 449)

File: src/server/controllers/users.controller.js [hunk 4]
```diff
@@ -462,7 +499,6 @@ export function editGeneralProfile(req, res, next) {
         res.status(200).json({ msg: userCallback.SUCCESS_UPDATE_WITHOUT_IMAGE });
       } else {
         const S3 = new AWS.S3({ region: 'ap-northeast-2' });
-        const bucketName = 'yodabucket';
         let now = new Date();
```
Comment (MechanicKim) [inline src/server/controllers/users.controller.js]: 'now', 'imageKey', 'encondedImage' are never reassigned. Use 'const' instead

File: src/server/controllers/users.controller.js [hunk 5]
```diff
@@ -476,8 +512,8 @@ export function editGeneralProfile(req, res, next) {
         S3.putObject(params).promise()
           .then((data, err) => {
             if (data) {
-              let profileUrl = `${S3.endpoint.href}${bucketName}/${imageKey}`;
-              return updateProfile(req, profileUrl);
+              let profileUrl = `${S3_endpoint_href}${bucketName}/${imageKey}`;
```
Comment (MechanicKim) [inline src/server/controllers/users.controller.js]: 'profileUrl' is never reassigned. Use 'const' instead
'profileUrl' is assigned a value but never used

File: src/server/controllers/users.controller.js (hunk 4 above)
Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: Thanks!

File: src/server/controllers/users.controller.js (hunk 5 above)
Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: Thanks! profileUrl was used in line 487!

File: src/server/controllers/users.controller.js (hunk 1 above)
Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: Thanks. Will change as `S3EndpointHref`

File: src/server/controllers/users.controller.js (hunk 2 above)
Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: Can you explain more about this?

**Comment by KyuaKwon** [comment]
Are you waiting for approval?

**Comment by lyw0149** [comment]
LGTM

**Pull request merged**